src/npyffi/flags.rs

```diff
@@ -71,7 +71,7 @@ pub const NPY_NEEDS_INIT: npy_char = 0x08;
 pub const NPY_NEEDS_PYAPI: npy_char = 0x10;
 pub const NPY_USE_GETITEM: npy_char = 0x20;
 pub const NPY_USE_SETITEM: npy_char = 0x40;
-pub const NPY_ALIGNED_STRUCT: npy_char = -128; // 0x80
+pub const NPY_ALIGNED_STRUCT: npy_char = 0x80_u8 as npy_char;
```

Comment:
`0x80 != -128`. `0x80 == 128`. So if the value was supposed to be `128`, then there is no need to cast to npy_char because the value is positive. Assigning to `0x80` should work.

Reply:
While `0x80 == 128`, this was values does not fit into the range of `i8` which is `-128..127`, so assigning `0x80` to an `i8` does not work, c.f. https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c48403c877eae77a660189bc05aca984

But `0x80_u8` does have the same bit pattern `1000_0000` as `-128_i8` (as the error message in the linked playground indicates) which is why it was written that way. So starting with `0x80_u8`, the cast will be a no-op for platforms where `c_char == u8` and will yield the same result as before on platforms where `c_char == i8`.

Reply:
Linking to the playground, I think selectively disabling the `overflowing_literals` lint is probably nicer as it is more self-documenting. Could you give the amended commit a try as well? Thanks!

Reply:
The latest builds locally without any problem. Looking forward to seeing it published. Thanks.